Automattic/newspack-custom-content-migrator
# Add NCC trickery

This adds support for args in the block encode/decode commands:

```
  [--post-id=<post-id>]
    The ID of the post to process.

  [--num-items=<num-items>]
    The number of posts to process.

  [--min-post-id=<min-post-id>]
    The minimum post ID to process.

  [--max-post-id=<max-post-id>]
    The maximum post ID to process.

  [--post-types=<post-types>]
    Comma-separated list of post types to process. Default is "post".
```

There is a new command that "nudges" posts to be picked up by the NCC - it just inserts a line break at the top of posts that start with `<!--` that would otherwise not get picked up by the NCC.


## How to test

* Make sure the NCC is installed and activated
* Run `wp newspack-content-migrator transform-blocks-encode --num-items=50 `. The output will tell you next steps. It's also logged to file.
* To test the "nudger", run this for example: `wp newspack-content-migrator transform-blocks-nudge --num-items=30  --post-types=post,page`

---

- [x] confirmed that PHPCS has been run – yea, kinda!
